### ISSUE TITLE
Licensing Considerations for Extension: Qreacto

### ISSUE THREAD

**Issue by dweng0** [clearmatics/qreacto#1]
**TL;DR**
I propose we use an MIT licensing strategy for this extension, and would like your thoughts and feedback.

Hi @aiman  @flofrie
As part of the process of migrating over to [Quarto](https://quarto.org/), I have developed an extension for Quarto that allows us to inject React components into quarto markdown documentation, I think it is something the Quarto community would love to be able to use and so we need to consider the licensing strategy for this extension. 

To begin, let's go through its functionality. 
Qreacto enables the injection of React components into markdown documents within a Quarto project. The user creates a react component in javascript or typescript and provides the necessary shortcode in the md document. The extension will find, inject, transpile and execute the component, providing all the reactivity of a typical React component.

When reviewing peer extensions, I noticed that they generally adopt the MIT license. To foster collaboration within the Quarto community, it would be prudent to follow a similar licensing strategy. However, i'm not sure what the best approach is and would lilke your thoughts and feedback on the matter. 

Once the licensing is in place I would like the repository to be made public. This will be beneficial for the continued development of the extension becase any [questions or issues](https://github.com/quarto-dev/quarto-cli/discussions) raised over at Quarto's repo typically need to have a repo linked for any meaningful responses. Again happy to discuss the pros and cons of this.

The most popular extensions in the eco system all use MIT as at 28th July 2023
- [Sverto](https://github.com/jimjam-slam/sverto/blob/main/LICENSE)
- [Light box](https://github.com/quarto-ext/lightbox/blob/main/LICENSE)
- [Social share](https://github.com/schochastics/quarto-social-share/blob/main/LICENSE)
- [Font awesome](https://github.com/quarto-ext/fontawesome/blob/main/LICENSE) 

to name a few
